### PR TITLE
Docs/986/move wiki to gh pages

### PR DESCRIPTION
**Pull request by NearW**
# Moved developer guides to gh-pages

_**Please read the [CONTRIBUTING.md](https://github.com/MaibornWolff/codecharta/blob/main/CONTRIBUTING.md) before opening a PR.**_

Issue: #986 

## Description

- Move developer guides to gh-pages
- Fix old dockerhub link
- Update copyright
- Fix broken image links
- Update the `new-to-code` doc with the `getting started` dev-guide wiki entry
- Deleted wiki entries

### CODE REVIEW

File: gh-pages/_posts/dev-guide/2020-07-31-e2e-testing-with-puppeteer.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  launch: {
+    headless: true,
+    args: ["--allow-file-access-from-files", "--start-maximized"],
+    defaultViewport: { width: 1920, height: 1080 },
+    // slowMo: 250
+  },
+};
+```
+
+- Pipe console.logs from the browser to the git-bash (Run `enableConsole()` from the `puppeteer.helper.ts` in a test)
```
Comment (BridgeAR) [inline gh-pages/_posts/dev-guide/2020-07-31-e2e-testing-with-puppeteer.md:53]:
```suggestion
- Pipe console.logs from the browser to the terminal (Run `enableConsole()` from the `puppeteer.helper.ts` in a test).
```

File: gh-pages/_posts/dev-guide/2020-07-31-e2e-testing-with-puppeteer.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+- Trying to access a selector that is not available (yet)
+- Accessing an old selector and using the data to verify something (reading old classNames and expecting a new className to be set)
+- Fixed delays by calling `waitFor(1000) // wait for 1000ms`
+
+**Best practices**
+
+- Before accessing a selector, wait until it's available using `await page.waitForSelector(MY_SELECTOR)`
+- When clicking a button, use `expect(page).toClick(MY_SELECTOR, { timeout: 3000 })`. This function awaits the selector for 3s before throwing an error
+- After clicking a button or changing the state, use `waitForSelector()` to verify, that the new state is rendered before continueing
```
Comment (BridgeAR) [inline gh-pages/_posts/dev-guide/2020-07-31-e2e-testing-with-puppeteer.md:124]:
```suggestion
- After clicking a button or changing the state, use `waitForSelector()` to verify, that the new state is rendered before continuing
```

File: gh-pages/_docs/07-01-new-to-code.md [hunk 3]
```diff
@@ -3,45 +3,151 @@ permalink: /docs/new-to-code/
...
+
+### Sonar
+
+- Publishes Sonar-Analysis-Results to Sonarcloud.io
+
+### Deploy
+
+- Deploys the application in a docker container to the github-pages
+- Publishes the new version on npm
+- Publishes a docker container on Docker Hub (not working atm)
```
[the remark this answers is not in the document]
Comment (NearW) [reply gh-pages/_docs/07-01-new-to-code.md:34]: It was. Nice catch. Didnt read through the whole document 👍

File: gh-pages/_docs/07-01-new-to-code.md [hunk 4]
```diff
@@ -3,45 +3,151 @@ permalink: /docs/new-to-code/
...
+
+We create Pull Requests to master after implementing a feature or fixing a bug. There is no release or development branch. We never push on master directly. Please take a look at our [contributing guidelines](https://github.com/MaibornWolff/codecharta/blob/master/CONTRIBUTING.md) before you start comitting.
+
+# Travis CI
+
+In Travis CI, we defined stages, which group different jobs. Inside a stage, all jobs run in parallel. There is no data persistence between stages, so we have to rebuild our application in each stage. The CI consists of the following stages:
+
+- Testing (which runs all the time)
+- Sonar (only run when a PR was merged with master)
+- Deploy (only run when executing the make_release.sh)
```
[the remark this answers is not in the document]
Comment (NearW) [reply gh-pages/_docs/07-01-new-to-code.md:20]: it's actually
```md
- Deploy (run by `make_release.py`)
```

File: gh-pages/_docs/07-01-new-to-code.md [hunk 5]
```diff
@@ -3,45 +3,151 @@ permalink: /docs/new-to-code/
...
 
-Finally it would be great if you looked at how we give and receive [feedback]({{site.baseurl}}{% link _docs/01-07-feedback.md %}). If you want to contribute you should check our [CONTRIBUTING.md](https://github.com/MaibornWolff/codecharta/blob/main/CONTRIBUTING.md) and see what naming conventions branches and commit messages use.
+Finally, it would be great if you looked at how we give and receive [feedback]({{site.baseurl}}{% link _docs/01-07-feedback.md %}) and after finishing this entire post, please explore our [developer guides]({{site.baseurl}}/categories/#dev-guide).
+
+# Branching / Releasing
+
+We create Pull Requests to the `main` branch after implementing a feature or fixing a bug. There is no release or development branch. We never push on `main` directly. Please take a look at our [contributing guidelines](https://github.com/MaibornWolff/codecharta/blob/master/CONTRIBUTING.md) before you start comitting.
```
Comment (NearW) [inline gh-pages/_docs/07-01-new-to-code.md:12]: replace master with main

File: gh-pages/_docs/07-01-new-to-code.md [hunk 6]
```diff
@@ -3,45 +3,151 @@ permalink: /docs/new-to-code/
...
+
+### Importing the Project
+
+Don't import the whole codecharta project to IntelliJ when working on the analysis. Simply import the analysis folder for that. IntelliJ might not able to identify it as a Gradle project otherwise.
+
+### Definitions
+
+#### Importer
+
+- Retrieves metrics from an external source, such as SonarQube and creates a cc.json
```
[the remark this answers is not in the document]
Comment (NearW) [reply gh-pages/_docs/07-01-new-to-code.md:48]:
```
- Retrieves metrics from external sources, such as `SonarQube` and creates the `cc.json`.
```

File: gh-pages/_posts/dev-guide/2020-07-31-e2e-testing-with-puppeteer.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+[Puppeteer](https://github.com/puppeteer/puppeteer) is a headless Chrome Node.js API to execute actions on a wegpage and assert queries using jest. The tests can be run headless (in the background) or headful (opening a browser window and executing the actions live). Puppeteer provides an API that lets you select elements and interact with the html-template through selectors and events. Those actions are wrapped in a page-object (.po.ts)
+
+# Running e2e tests
+
+When you run e2e tests, puppeteers starts a html-server using the built application as it's entry point. In order to run the e2e tests, the application has to be built. Keep in mind that changing code in files, which change the outcome of the build, will require a rebuild of the application to test the new code. This includes all css, html and mostly all .ts-files (page-objects and test-files are excluded from the build and don't require a rebuild).
+
+### IMPORTANT
+
+- Do not run the dev-server while running e2e tests. They share the same build-folder and e2e tests are not executable from a dev-server compiled build.
+- Do not run the tests in a console included in the IDE. I recommend using the git-bash.
```
[the remark this answers is not in the document]
Comment (NearW) [reply gh-pages/_posts/dev-guide/2020-07-31-e2e-testing-with-puppeteer.md:22]: They're working as long as you run the tests headless. Headful tests can't be executed in the IntelliJ-Terminal at least.
I will check this again to be sure.

[the remark this answers is not in the document]
Comment (NearW) [reply gh-pages/_posts/dev-guide/2020-07-31-e2e-testing-with-puppeteer.md:22]: Works

**Pull request merged**